tests/suite-unit/PluginFlyvemdmPolicyRemoveapplication.php

```diff
@@ -0,0 +1,130 @@
+ * modify it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * Flyve MDM Plugin for GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Flyve MDM Plugin for GLPI. If not, see http://www.gnu.org/licenses/.
+ * ------------------------------------------------------------------------------
+ * @author    Domingo Oropeza
+ * @copyright Copyright © 2018 Teclib
+ * @license   AGPLv3+ http://www.gnu.org/licenses/agpl.txt
+ * @link      https://github.com/flyve-mdm/glpi-plugin
+ * @link      https://flyve-mdm.com/
+ * ------------------------------------------------------------------------------
+ */
+
+namespace tests\units;
+
+use Glpi\Test\CommonTestCase;
+
+class PluginFlyvemdmPolicyRemoveapplication extends CommonTestCase {
+
+   private $dataField = [
+      'group'     => 'application',
+      'symbol'    => 'removeApp',
+      'type_data' => '',
+      'unicity'   => 0,
+   ];
+
+   private $packageName = 'com.domain.author.application.apk';
+   private $filename = 'application.apk';
+
+   /**
+    * @return array
+    */
+   private function createNewPolicyInstance() {
+      $policyData = new \PluginFlyvemdmPolicy();
+      $policyData->fields = $this->dataField;
+      $policy = $this->newTestedInstance($policyData);
+      return [$policy, $policyData];
+   }
+
+   protected function validationProvider() {
+      return [
+         'Check ID exist'            => [
+            'data'     => ['', '', 0],
+            'expected' => [false, 'An application ID is required'],
+         ],
+         'Check itemtype is empty 1' => [
+            'data'     => ['lorem', 'Bad Value', 0],
+            'expected' => [false],
+         ],
+         'Check itemtype is empty 2' => [
+            'data'     => ['lorem', '', 1],
+            'expected' => [false],
+         ],
+         'Valid check 1'             => [
+            'data'     => ['any value', '', 0],
+            'expected' => [true],
+         ],
+      ];
+   }
+
+   /**
+    * @dataProvider validationProvider
+    * @tags testIntegrityCheck
+    * @param $data
+    * @param $expected
+    */
+   public function testIntegrityCheck($data, $expected) {
+      list($policy) = $this->createNewPolicyInstance();
+      $success = $policy->integrityCheck($data[0], $data[1], $data[2]);
+      $this->boolean($success)->isEqualTo($expected[0]);
+      if (!$expected[0] && isset($expected[1])) {
+         $this->string($_SESSION["MESSAGE_AFTER_REDIRECT"][0][0])->isEqualTo($expected[1]);
+         unset($_SESSION["MESSAGE_AFTER_REDIRECT"]); // to clear the buffer
```

Comment:
As you often use beforeTestMethod() to setup things, I think you should use afterTestMethod() to unset this value.

Reply:
then OK